src/test/java/co/zeroae/gate/AppTest.java

```diff
@@ -83,18 +83,31 @@ public void testMissingContentType() {
 
     @Test
     public void testGateJSONResponse() throws Exception {
-        input_headers.put("Accept", "application/gate+json");
+        for (String responseType: new String[]{"application/gate+json", "application/json"}) {
+            input_headers.put("Accept", responseType);
+
+            final APIGatewayProxyResponseEvent result = app.handleRequest(input, context);
+            assertEquals(200, result.getStatusCode().intValue());
+
+            // Ensure we get back application/gate+json back
+            assertEquals(responseType, result.getHeaders().get("Content-Type"));
+            final JsonFactory factory = new JsonFactory();
+            final JsonParser parser = factory.createParser(result.getBody());
+            while (!parser.isClosed()) {
+                parser.nextToken();
+            }
+        }
+    }
+
+    @Test
+    public void testFastInfosetResponse() {
+        input_headers.put("Accept", "application/fastinfoset");
 
         final APIGatewayProxyResponseEvent result = app.handleRequest(input, context);
         assertEquals(200, result.getStatusCode().intValue());
 
-        // Ensure we get back application/gate+json back
-        assertEquals("application/gate+json", result.getHeaders().get("Content-Type"));
-        final JsonFactory factory = new JsonFactory();
-        final JsonParser parser = factory.createParser(result.getBody());
-        while (!parser.isClosed()) {
-            parser.nextToken();
-        }
+        // Ensure we get back application/fastinfoset back
+        assertEquals("application/fastinfoset", result.getHeaders().get("Content-Type"));
```

Comment:
ensure we check the content is base64 encoded.

ref: https://docs.aws.amazon.com/apigateway/latest/developerguide/lambda-proxy-binary-media.html